### ISSUE TITLE
Dataset/dataloader API

### ISSUE THREAD

**Issue by vvmnnnkv** [OpenMined/syft.js#214]
## Feature Description
Add helper class that to make easier data loading, batching, shuffling, etc.
I.e. dataset and/or dataloader class API that would help user to prepare data for training.

Use cases: 
* Easify and standardize data preparation on the client side, across all libraries.
* Supply dataset into `Job.train(..., {data: dataset, ...})` and update Job.train to work with the dataset API.

Design: TBD

## Is your feature request related to a problem?
Standardize the way data is loaded into training loop across workers.

## What alternatives have you considered?
Discussed with FL team.

## Additional Context
See also #212
